Mathlib/NumberTheory/ModularForms/EisensteinSeries/UniformConvergence.lean

```diff
@@ -0,0 +1,311 @@
+    add_zero, normSq_apply, add_re, mul_re, sub_zero, add_im] at *
+  exact H1
+
+lemma auxbound2 (z : ℍ) (δ : ℝ) {ε : ℝ} (hε : 1 ≤ ε ^ 2) : r z ≤ Complex.abs (δ * (z : ℂ) + ε) := by
+  rw [r, Complex.abs, min_le_iff]
+  have H1 : Real.sqrt (r1 z) ≤ Real.sqrt ((δ * (z : ℂ).re + ε) * (δ * (z : ℂ).re + ε) +
+      δ * (z : ℂ).im * (δ * (z : ℂ).im)) := by
+    rw [r1, Real.sqrt_le_sqrt_iff, ← pow_two, ← pow_two]
+    apply r1_aux_bound z δ hε
+    nlinarith
+  right
+  simp only [ne_eq, coe_re, coe_im, normSq_apply, AbsoluteValue.coe_mk, MulHom.coe_mk, add_re,
+    mul_re, ofReal_re, ofReal_im, zero_mul, sub_zero, int_cast_re, add_im, mul_im, add_zero,
+    int_cast_im] at *
+  exact H1
+
+lemma ne_zero_if_max {x : Fin 2 → ℤ} (hx : x ≠ 0)
+    (h : (max (x 0).natAbs (x 1).natAbs) = (x 0).natAbs) : (x 0) ≠ 0 := by
+  intro h0
+  rw [h0] at h
+  simp only [ne_eq, Int.natAbs_zero, ge_iff_le, zero_le, max_eq_right, Int.natAbs_eq_zero] at *
+  have : x = ![x 0, x 1] := List.ofFn_inj.mp rfl
+  rw [h0, h] at this
+  simp only [this, Matrix.cons_eq_zero_iff, Matrix.zero_empty, and_self, not_true_eq_false] at hx
+
+lemma ne_zero_if_max' {x : Fin 2 → ℤ} (hx : x ≠ 0)
+    (h : (max (x 0).natAbs (x 1).natAbs) = (x 1).natAbs) : (x 1) ≠ 0 := by
+  apply ne_zero_if_max (x := ![x 1, x 0]) ?_ (by simpa using h)
+  simp only [ne_eq, Matrix.cons_eq_zero_iff, Matrix.zero_empty, and_true, not_and]
+  intro h1 h0
+  rw [Function.ne_iff, Fin.exists_fin_two, h1, h0] at hx
+  simp only [Pi.zero_apply, ne_eq, not_true_eq_false, or_self] at *
+
+lemma div_max_sq_ge_one (x : Fin 2 → ℤ) (hx : x ≠ 0) :
+    (1 : ℝ) ≤ (x 0 / (max (x 0).natAbs (x 1).natAbs)) ^ 2 ∨
+      (1 : ℝ) ≤ (x 1 / (max (x 0).natAbs (x 1).natAbs)) ^ 2 := by
+  cases' (max_choice (x 0).natAbs (x 1).natAbs) with H1 H2
+  · left
+    rw [H1, div_pow, Int.cast_natAbs (x 0), Int.cast_abs]
+    have : (x 0 : ℝ) ≠ 0 := by
+      simpa using (ne_zero_if_max hx H1)
+    have h1 : (x 0 : ℝ) ^ 2/(_root_.abs (x 0 : ℝ)) ^ 2 = 1 := by
+      simp only [_root_.sq_abs, ne_eq, OfNat.ofNat_ne_zero, not_false_eq_true, pow_eq_zero_iff,
+        this, div_self]
+    exact h1.symm.le
+  · right
+    rw [H2,div_pow, Int.cast_natAbs (x 1), Int.cast_abs]
+    have : (x 1 : ℝ) ≠ 0 := by
+      simpa using (ne_zero_if_max' hx H2)
+    have h1 : (x 1 : ℝ)^2/(_root_.abs (x 1 : ℝ))^2 = 1 := by
+      simp only [_root_.sq_abs, ne_eq, OfNat.ofNat_ne_zero, not_false_eq_true, pow_eq_zero_iff,
+        this, div_self]
+    exact h1.symm.le
+
+lemma rpow_bound {k : ℝ} (hk : 0 ≤ k) (z : ℍ) (x : Fin 2 → ℤ) (hx : x ≠ 0) :
+    ((r z) ^ k) * (max (x 0).natAbs (x 1).natAbs) ^ k ≤
+      (Complex.abs ((x 0 : ℂ) * (z : ℂ) + (x 1 : ℂ))) ^ k := by
+  by_cases hk0 : k ≠ 0
+  · let n := max (x 0).natAbs (x 1).natAbs
+    have hn0 : n ≠ 0 := by
+      rw [← Iff.ne ((mem_box_eq_zero_iff_eq_zero (α := ℤ × ℤ)) (x 0, x 1) (by simp)),
+        ← Iff.ne (Function.Injective.eq_iff (Equiv.injective (piFinTwoEquiv fun x ↦ ℤ)))] at *
+      simpa using hx
```

Comment:
I'm not sure it's better, but this also works:
```suggestion
      rw [← pos_iff_ne_zero]
      cases max_choice (x 0).natAbs (x 1).natAbs with
      | inl h => exact lt_max_of_lt_left <| Int.natAbs_pos.mpr <| ne_zero_if_max hx h
      | inr h => exact lt_max_of_lt_right <| Int.natAbs_pos.mpr <| ne_zero_if_max' hx h
```

Reply:
its more lines, so I'll stick with the original one if thats ok.